### PR TITLE
[build] More tweaks after the removal of Makefile build infrastructure

### PR DESCRIPTION
**Pull request by ejgallego**
Mainly cleanup of configure values that are not used anymore, and important fixes to `doc/README.md`

### CODE REVIEW

**Comment by Zimmi48** [comment]
`Makefile.docgram` is currently included in `Makefile.dune` to provide the docgram targets. If it is removed, its rules will need to be reimplemented by coq_dune, which might be complicated. IMHO it is simpler to just accept that some edge things still use a Makefile for now (less work for you, and less review work for us).

The doc install targets might also turn out to be hard to implement with Dune < 3.0, so using a Makefile for the documentation targets in general might be a good idea for the time being.

**Comment by ejgallego** [comment]
Thanks for the feedback @Zimmi48 , I may try to have gen_rules take care of those too, not sure whether they'll be doable, but if we can get rid of all make, IMHO better.

File: tools/configure/configure.ml [hunk 1]
```diff
@@ -209,7 +204,7 @@ let check_for_zarith prefs =
 let check_sphinx_deps () =
   ignore (run (which "python3") ["doc/tools/coqrst/checkdeps.py"])
 
-let check_doc () =
+let _check_doc () =
```
Comment (SkySkimmer) [inline tools/configure/configure.ml:207]: Why keep?

Comment (ejgallego) [reply tools/configure/configure.ml:207]: As a reminder to think where we should perform this check.

Comment (ejgallego) [reply tools/configure/configure.ml]: Ok, fixed and fixed stale docs about docs.

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**